### PR TITLE
test(longevity): Many client with big cluster

### PR DESCRIPTION
**Pull request by soyacz**
Longevity many client tests
Jenkins staging job: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/longevity-many-clients-4h-test/
trello: https://trello.com/c/HZPKYzt9/705-scale-longevity-test-case-with-many-clients
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] ~I added the relevant `backport` labels~
- [ ] ~New configuration option are added and documented (in `sdcm/sct_config.py`)~
- [ ] ~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~
- [ ] All new and existing unit tests passed (CI)
- [ ] ~I have updated the Readme/doc folder accordingly (if needed)~

### CODE REVIEW

**Comment by roydahan** [comment]
Did you  run a  test in staging?
Does it pass?
Please share a monitor link here if you have it.

File: test-cases/longevity/longevity-many-clients-4h.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+test_duration: 360
+
+prepare_write_cmd:  "cassandra-stress write cl=QUORUM n=20971520 -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=80 -pop seq=1..20971520 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5"
+
+stress_cmd: ["cassandra-stress write cl=QUORUM duration=240m -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=40 -pop seq=1..20971520 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5",
+             "cassandra-stress read  cl=QUORUM duration=240m -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=40 -pop seq=1..20971520 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5"
```
Comment (roydahan) [inline test-cases/longevity/longevity-many-clients-4h.yaml:6]: I think you'll need to reduce the amount of threads.
If you have 50 client each one will run 2  commands with 40 threads each, the load may be too high.

File: test-cases/longevity/longevity-many-clients-4h.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+nemesis_class_name: 'ChaosMonkey'
+nemesis_interval: 5
+nemesis_filter_seeds: false
+seeds_num: 1
+
+user_prefix: 'longevity-many-clients-4h'
+space_node_threshold: 64424
+
+pre_create_schema: True
+sstable_size: 100
```
[the remark this answers is not in the document]
Comment (soyacz) [reply test-cases/longevity/longevity-many-clients-4h.yaml]: removed

File: test-cases/longevity/longevity-many-clients-4h.yaml (hunk 1 above)
Comment (soyacz) [reply test-cases/longevity/longevity-many-clients-4h.yaml]: Reduced. This was taken from longevity 4h. I did some math now, and assuming on longevity-4h was around 1.67 stress thread per scylla node core (2 loaders x 80 threads / 96 cores = 1,67 threads/core), I kept the similar ratio for this test resulting in 7 thread (50 loaders x 7 threads / 216 cores = 1,62 threads/core). Let me know if this thinking is ok.

**Comment by fgelcer** [comment]
@roydahan , i was working with @soyacz to check the current status of a run with this PR, and we think the load is too low...
Lukasz will be increasing it so we will re-run..

also, because of the prepare now will be faster, we maybe will be able to increase the population (and the `n=` in the c-s command), WDYT?

here some screenshots about the start of the load:

Screenshot from 2021-11-09 11-09-31

**Comment by fgelcer** [comment]
once you are ready with it, move it to "ready for review"

**Comment by soyacz** [comment]
having 50 keyspaces_num: 50 and round_robin: true, load is very high (75% cpu, disk bound) and data volume reaching almost 9TB after 2,5h. Before the end of the test, cluster became overloaded and performance dropped significantly.

I've decreased col `n` param from `10` to `5` to decrease volume and disk utilization for the next execution.

Link to the most recent run: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/longevity-many-clients-4h-test/11/ (test id: d4ab9d5e-eb47-43f8-92b6-258f07c807d1).

**Comment by roydahan** [comment]
What is the latest result of this run?
I know we had a mail thread about changing some parameters, but what's the final outcome?

**Review by roydahan — APPROVED** [review]
I saw the latest results in the mail thread, it looks good.

**Pull request merged**